File: fe/lib/models/lesson.dart
import 'package:fe/models/lesson_material.dart';

class Lesson {
  String id;
  String title;
  DateTime createdAt;
  int level;
  double progress;
  List<String> materialIds;

  Lesson(
      {required this.createdAt,
      required this.id,
      required this.title,
      required this.level,
      required this.progress,
      required this.materialIds});

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      id: json["_id"] as String,
      title: json["title"] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
      level: json['level'] as int,
      progress: json['progress'] as double,
      materialIds: List<String>.from(json['material'] ?? []),
    );
  }
}

File: be/models/lesson.js
const mongoose = require("mongoose");

const lessonSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  createdAt: { type: Date, default: Date.now },
  level: Number,
  material: [{ type: mongoose.Schema.Types.ObjectId, ref: "Lesson_Material" }],
  progress: { type: Number, default: 0 },
});

const Lesson = mongoose.model("Lesson", lessonSchema);

module.exports = Lesson;

const lessonDumy = [
  {
    title: "Inggris Beginner",
    level: 1,
  },
  {
    title: "Inggris Medium",
    level: 2,
  },
  {
    title: "Inggris Hard",
    level: 2,
  },
];

// mongoose.connect(
//   "mongodb+srv://motherbloodss:XKFofTN9qGntgqbo@cluster0.ejyrmvc.mongodb.net/?retryWrites=true&w=majority",
//   {
//     useNewUrlParser: true,
//     useUnifiedTopology: true,
//   }
// );

// const seedLessons = async () => {
//   try {
//     await Lesson.deleteMany({});
//     await Lesson.insertMany(lessonDumy);
//     console.log("Data dummy berhasil dimasukkan ke database!");
//     mongoose.connection.close();
//   } catch (error) {
//     console.error("Error saat memasukkan data dummy:", error);
//   }
// };

// seedLessons();
